plugins/common/http/config.go

```diff
@@ -52,7 +53,14 @@ func (h *HTTPClientConfig) CreateClient(ctx context.Context, log telegraf.Logger
 		Timeout:   time.Duration(timeout),
 	}
 
-	client = h.OAuth2Config.CreateOauth2Client(ctx, client)
+	h.OAuth2Config.TokenURL = fmt.Sprintf("%v", ctx.Value("url"))
+
+	client, err = h.OAuth2Config.CreateOauth2Client(ctx, client)
+	if err != nil {
+		return nil, err
```

Comment:
Please return some context for the error message to be able to traceback to this call..

(same for all other `return err` calls)